src/Symfony/Component/String/AbstractUnicodeString.php

```diff
@@ -135,15 +135,21 @@ public function ascii(array $rules = []): self
             } elseif (!\function_exists('iconv')) {
                 $s = preg_replace('/[^\x00-\x7F]/u', '?', $s);
             } else {
-                $s = @preg_replace_callback('/[^\x00-\x7F]/u', static function ($c) {
-                    $c = (string) iconv('UTF-8', 'ASCII//TRANSLIT', $c[0]);
-
-                    if ('' === $c && '' === iconv('UTF-8', 'ASCII//TRANSLIT', '²')) {
-                        throw new \LogicException(sprintf('"%s" requires a translit-able iconv implementation, try installing "gnu-libiconv" if you\'re using Alpine Linux.', static::class));
-                    }
+                $previousLocale = setlocale(\LC_CTYPE, 0);
+                try {
+                    setlocale(\LC_CTYPE, 'C');
```

Comment:
whta about this?
```suggestion
                    setlocale(\LC_CTYPE, 'C.UTF-8', 'C');
```

Reply:
or actually C.UTF-8 creates the issue?

Reply:
Exactly, `C.UTF-8` produces the wrong result.